From 6ede508ac7b890df01acd9cd219542d36c49b25e Mon Sep 17 00:00:00 2001
From: Vipul Pandya <vipul@chelsio.com>
Date: Thu, 10 May 2012 15:55:10 +0530
Subject: [PATCH 07/14] cxgb3: Revert implement set_phys_id

This patch is a reverse patch of the following upstream commit
12fcf941674fd781117a56f998d2bb28b4bc4cf1

Signed-off-by: Vipul Pandya <vipul@chelsio.com>
---
 drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c |   24 ++++++++++------------
 1 files changed, 11 insertions(+), 13 deletions(-)

diff --git a/drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c b/drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c
index d9ee262..a671d43 100644
--- a/drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb3/cxgb3_main.c
@@ -1728,26 +1728,24 @@ static int restart_autoneg(struct net_device *dev)
 	return 0;
 }

-static int set_phys_id(struct net_device *dev,
-		       enum ethtool_phys_id_state state)
+static int cxgb3_phys_id(struct net_device *dev, u32 data)
 {
 	struct port_info *pi = netdev_priv(dev);
 	struct adapter *adapter = pi->adapter;
+	int i;

-	switch (state) {
-	case ETHTOOL_ID_ACTIVE:
-		return 1;	/* cycle on/off once per second */
-
-	case ETHTOOL_ID_OFF:
-		t3_set_reg_field(adapter, A_T3DBG_GPIO_EN, F_GPIO0_OUT_VAL, 0);
-		break;
+	if (data == 0)
+		data = 2;

-	case ETHTOOL_ID_ON:
-	case ETHTOOL_ID_INACTIVE:
+	for (i = 0; i < data * 2; i++) {
 		t3_set_reg_field(adapter, A_T3DBG_GPIO_EN, F_GPIO0_OUT_VAL,
-			 F_GPIO0_OUT_VAL);
+				 (i & 1) ? F_GPIO0_OUT_VAL : 0);
+		if (msleep_interruptible(500))
+			break;
 	}

+	t3_set_reg_field(adapter, A_T3DBG_GPIO_EN, F_GPIO0_OUT_VAL,
+			 F_GPIO0_OUT_VAL);
 	return 0;
 }

@@ -2063,7 +2061,7 @@ static const struct ethtool_ops cxgb_ethtool_ops = {
 	.set_pauseparam = set_pauseparam,
 	.get_link = ethtool_op_get_link,
 	.get_strings = get_strings,
-	.set_phys_id = set_phys_id,
+	.phys_id = cxgb3_phys_id,
 	.nway_reset = restart_autoneg,
 	.get_sset_count = get_sset_count,
 	.get_ethtool_stats = get_stats,
--
1.7.1

